### DISCUSSION TITLE
How to pass arrays of Autodesk types?

### DISCUSSION THREAD

**Discussion by rchive1** [chuongmep/CadPythonShell#9] [Q&A] [answered]
I'm trying to set a TIN surface's elevation colors with Surface.Analysis.SetElevationData() which requires an Array of type SurfaceAnalysisElevationData. It won't accept a python list or a single SurfaceAnalysisElevationData. How can I create an Array of this type, or of any Autodesk data type?

**Comment by chuongmep** [comment]
Hi @rchive1 , you can try reference code C# code here : 
https://github.com/15831944/EM/blob/0513d2047d7a5cdcac917906b190eabf48db25e9/Base/Base_Tools45/Base_Tools45/C3D/Surf_Analysis.cs#L49

**Reply by rchive1** [reply]
In that definition of `CreateElevationRegions()`, there is:
`SurfaceAnalysisElevationData[] newData = new SurfaceAnalysisElevationData[steps];`
How do you translate that to Python?

**Reply by rchive1** [reply, marked as answer]
I figured it out.
The translation of C#:
`SurfaceAnalysisElevationData[] newData = new SurfaceAnalysisElevationData[steps];`
to Python seems to be:
`newData = System.Array[SurfaceAnalysisElevationData](steps)`

I've never used Python with .NET or IronPython before, so I couldn't figure out what "Array" was referring to or how to create one.

Thanks for your help.
